### PR TITLE
First stab at server initiated key rotation

### PR DESCRIPTION
**Pull request by grahamgilbert**

### CODE REVIEW

File: Package/checkin [hunk 1]
```diff
@@ -133,14 +134,40 @@ def escrow_key(plist):
...
+        # Don't do anything if we don't care about the good stuff
+        return ''
+
+    output_plist = pref('OutputPath')
+    if not os.path.exists(output_plist):
+        # Need this to be here too (which it should, but you never know..)
+        return ''
+
+    if json_output.get('rotation_required', False):
+        os.path.remove(output_plist)
```
Comment (weswhet) [inline Package/checkin]: Want to add a logging line here so we know the key was removed? Other than that LGTM

**Comment by grahamgilbert** [comment]
Alright, this is more complete. PTAL @weswhet

File: README.md [hunk 2]
```diff
@@ -81,6 +81,10 @@ As of version 3.0.0 you can now define the time interval in Hours for how often
 $ sudo defaults write /Library/Preferences/com.grahamgilbert.crypt KeyEscrowInterval -int 2
 ```
 
+### PostRunCommand
+
+(Introduced in version 3.2.0) This is a command that is run after Crypt has detected an error condition with a stored key that cannot be resolved silently - either it has failed validation or the server has instructed the client to rotate the key. These cannot be resolved silently on APFS volumes, so the user will need to log in again. If you have a tool that can enforce a logout or a reboot, you can run it here. This preference can either be a string if your command has no spaces, or an array if there are spaces in the command.
```
Comment (clburlison) [inline README.md:86]: Sorry for not testing this but I was under the impression that APFS could silently rotate the FV key. Should this section say HFS, or am I mistaken?

Comment (weswhet) [reply README.md:86]: How would you do this? To the best of knowledge, you can't use a recovery key to rotate the key on APFS.

**Comment by weswhet** [comment]
Looks good! nice work.

**Review by weswhet — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (clburlison) [reply README.md:86]: Thanks that gives me the context to realize I was confused. I was thinking of how you can rotate the APFS FV key while the encryption step is in process while HFS will fail.

Comment (clburlison) [reply README.md:86]: Only left over comment is adding a “defaults write” block to match the other steps or link to the mobileconfig to inform users how to enable this setting. 

```bash
$ defaults write  /Library/Preferences/com.grahamgilbert.crypt PostRunCommand -array ('/usr/local/munki/managedsoftwareupdate', '--auto') 
```

Untested code typed from my iPhone

**Pull request merged**